### DISCUSSION TITLE
How to create View component which width match Text inside it

### DISCUSSION THREAD

**Discussion by tomaszzmudzinski** [diegomura/react-pdf#1201] [General]
Hi,

I am trying to use View Component and Text Component to create something which looks like a button. The view component should always have the width of text inside it + padding given in styles. Don't know why, but the view component always has the width of the whole page.

![image](https://user-images.githubusercontent.com/4519335/114283483-4ae96800-9a4a-11eb-90a9-3c995305e924.png)

It works differently in raw HTML. Example below:

![image](https://user-images.githubusercontent.com/4519335/114283542-8a17b900-9a4a-11eb-95eb-4acc2ad1c23d.png)

Could anyone tell me how to do it? I can't set width manually because the text is dynamic so its width can change.

**Comment by daihuaye** [comment]
One possible way to do is use canvas.measureText method to calculate the width and use the width as inline CSS to style the View container. For example, here is sample code below:

``` javascript
const canvas = document.getElementById('canvas');
const ctx = canvas.getContext('2d');

let text = ctx.measureText('Hello world');
console.log(text.width);  // 56;

<View  style={{width: textwidth}}>
    <Text>Hello world</Text>
</View>
```

For your reference, here is the link to the measureText doc: https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/measureText

**Comment by benjick** [comment]
Here's how I did it

```tsx
const pillStyles = StyleSheet.create({
  pillContainer: {
    flexDirection: "row",
    alignItems: "center",
  },
  pill: {
    borderRadius: 12,
    paddingVertical: 4,
    paddingHorizontal: 8,
    alignSelf: "flex-start",
    border: "1px solid black",
  },
  pillText: {
    color: "black",
    fontSize: 8,
  },
});

const Pill = ({
  text,
  backgroundColor = "#ffb3d2",
}: {
  text: string;
  backgroundColor?: string;
}) => (
  <View style={pillStyles.pillContainer}>
    <View style={{ ...pillStyles.pill, backgroundColor }}>
      <Text style={pillStyles.pillText}>{text}</Text>
    </View>
  </View>
);
```
